pkg/secret/secret_test.go

```diff
@@ -1 +1,347 @@
+		},
+		{
+			builderNil:    false,
+			definitionNil: false,
+			apiClientNil:  true,
+			expectedError: "Secret builder cannot have nil apiClient",
+		},
+		{
+			builderNil:    false,
+			definitionNil: false,
+			apiClientNil:  false,
+			expectedError: "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder, _ := buildTestBuilderWithFakeObjects(nil, "test", "test")
+
+		if testCase.builderNil {
+			testBuilder = nil
+		}
+
+		if testCase.definitionNil {
+			testBuilder.Definition = nil
+		}
+
+		if testCase.apiClientNil {
+			testBuilder.apiClient = nil
+		}
+
+		result, err := testBuilder.validate()
+		if testCase.expectedError != "" {
+			assert.NotNil(t, err)
+			assert.Equal(t, testCase.expectedError, err.Error())
+			assert.False(t, result)
+		} else {
+			assert.Nil(t, err)
+			assert.True(t, result)
+		}
+	}
+}
+
+func TestSecretGet(t *testing.T) {
+	testCases := []struct {
+		secretExistsAlready bool
+		name                 string
+		namespace            string
+		secretType           string
+	}{
+		{
+			secretExistsAlready: true,
+			name:                 "test",
+			namespace:            "test",
+			secretType:           "test",
+			
+		},
+		{
+			secretExistsAlready: false,
+			name:                 "test",
+			namespace:            "test",
+			secretType:           "test",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		if testCase.secretExistsAlready {
+			runtimeObjects = append(runtimeObjects, &operatorv1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testCase.name,
+					Namespace: testCase.namespace,
+				},
+			})
+		}
+
+		testBuilder, _ := buildTestBuilderWithFakeObjects(runtimeObjects, testCase.name, testCase.namespace)
+
+		result, err := testBuilder.Get()
+		if testCase.secretExistsAlready {
+			assert.Nil(t, err)
+			assert.NotNil(t, result)
+		} else {
+			assert.NotNil(t, err)
+			assert.Nil(t, result)
+		}
+	}
+}
+
+func TestSecretExists(t *testing.T) {
+	testCases := []struct {
+		secretExistsAlready bool
+		name                 string
+		namespace            string
+	}{
+		{
+			secretExistsAlready: true,
+			name:                 "test",
+			namespace:            "test",
+		},
+		{
+			secretExistsAlready: false,
+			name:                 "test",
+			namespace:            "test",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		if testCase.secretExistsAlready {
+			runtimeObjects = append(runtimeObjects, &operatorv1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testCase.name,
+					Namespace: testCase.namespace,
+					Type:      testCase.secretType,
+				},
+			})
+		}
+
+		testBuilder, _ := buildTestBuilderWithFakeObjects(runtimeObjects, testCase.name, testCase.namespace, testCase.secretType)
+
+		result := testBuilder.Exists()
+		if testCase.secretExistsAlready {
+			assert.True(t, result)
+		} else {
+			assert.False(t, result)
+		}
+	}
+}
+
+
```

Comment:
Please add test cases for following functions: NewBuilder, WithOptions, WithAnnotations, WithData, Update